temporalio/worker/_workflow_instance.py

```diff
@@ -1646,8 +1743,20 @@ def _process_signal_job(
         input = HandleSignalInput(
             signal=job.signal_name, args=args, headers=job.headers
         )
+
+        async def run_signal():
+            try:
+                self._handled_signals_seq += 1
+                id = self._handled_signals_seq
+                self._in_progress_signals[id] = (job, defn)
+                await self._run_top_level_workflow_function(
```

Comment:
I am a bit concerned that this may introduce a new yield point which could affect ordering for past workflows. I can spend some time trying to think of ways to wrap signal calling without introducing a new yield point but still making sure interceptors are wrapped. Any initial ideas? Maybe a `add_done_callback` on the resulting created task?